.github/workflows/test-e2e-gpu.yaml

```diff
@@ -0,0 +1,89 @@
+name: GPU E2E Test
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
+
+jobs:
+  gpu-e2e-test:
+    name: GPU E2E Test
+    runs-on: oracle-vm-16cpu-a10gpu-240gb
+
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/kubeflow/trainer
+
+    strategy:
+      fail-fast: false
+      matrix:
+        kubernetes-version: ["1.33.1"]
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/kubeflow/trainer
+
+      - name: Check GPU label
+        id: check-label
+        run: |
+          if [[ "${{ join(github.event.pull_request.labels.*.name, ',') }}" != *"ok-to-test-gpu-runner"* ]]; then
+            echo "Label 'ok-to-test-gpu-runner' not found. Skipping GPU tests."
+            echo "skip=true" >> $GITHUB_OUTPUT
+          else
+            echo "Label found. Running GPU tests."
+            echo "skip=false" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Skip message
+        if: steps.check-label.outputs.skip == 'true'
+        run: echo "✅ Skipped GPU E2E tests (label not present)."
+
+      - name: Setup Go
+        if: steps.check-label.outputs.skip == 'false'
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: ${{ env.GOPATH }}/src/github.com/kubeflow/trainer/go.mod
+
+      - name: Setup Python
+        if: steps.check-label.outputs.skip == 'false'
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        if: steps.check-label.outputs.skip == 'false'
+        run: |
+          pip install papermill==2.6.0 jupyter==1.1.1 ipykernel==6.29.5
+          pip install git+https://github.com/kubeflow/sdk.git@main
+
+      - name: Delete any existing cluster
+        if: steps.check-label.outputs.skip == 'false'
+        run: |
+          make test-e2e-delete-gpu-cluster
```

Comment:
Can you delete the nvkind cluster as the final step of this workflow ?

Reply:
Hi @andreyvelich checking and deleting existing cluster is important, as there could be existing cluster running in the VM. I would suggest to keep this check and deleting step for existing cluster at initial phase of CI action. 
Do you want to delete AFTER the CI action **as well.** ?

Reply:
@jaiakash Can you just create finalize step in your workflow that will run:
```
kind delete cluster && echo "kind cluster has been deleted" || (echo "kind cluster doesn't exist" && exit 0)
```